examples/simple/igraph_assortativity_nominal.c

```diff
@@ -0,0 +1,44 @@
+#include <igraph.h>
+#include <stdio.h>
+
+int main() {
+    igraph_integer_t nodes = 250, types = 5;
+
+    igraph_matrix_t pref_matrix;
+    igraph_matrix_init(&pref_matrix, types, types);
+
+    igraph_rng_seed(igraph_rng_default(), 42);
+    printf("Randomly generated graph with 250 nodes and 5 vertex types\n\n");
+
+    /* Generate preference matrix giving connection probabilities for different vertex types */
+    for (igraph_integer_t i = 0; i < types; i++) {
+        for (igraph_integer_t j = i; j < types; j++) {
+            MATRIX(pref_matrix, i, j) = RNG_UNIF01();
+            MATRIX(pref_matrix, j, i) = MATRIX(pref_matrix, i, j);
+        }
+    }
+
+    for (int i = 0; i < 5; i++) {
+        igraph_t g;
+        igraph_vector_int_t node_type_vec;
+        igraph_real_t assortativity;
+        igraph_vector_int_init(&node_type_vec, nodes);
```

Comment:
Move the initialization/destruction of `node_type_vec` outside of the loop. This demonstrates that vectors can be re-used for a potential performance advantage.

Reply:
Hi. I tried to move the declarations outside the loop to reduce the runtime. But this introduces a few memory leak issues such as [this one](https://dev.azure.com/igraph-team/igraph/_build/results?buildId=3360&view=logs&j=4c3a8381-b9de-5463-904f-918810769be8&t=e71559a0-54d3-55f3-64cc-74b67ad70398&l=389). The issues seem to originate in lines 28 and 35 of the program. Is this because it is expected to have an uninitialized data structure at the beginning of every iteration of the loop?

Reply:
Only move the vector initialization/destruction outside (i.e. `node_type_vec`), not the graph.

`igraph_preference_game()` will create a new graph (i.e. it takes an uninitialized `igraph_t`), so it must stay inside of the loop.

Reply:
Oh, I think I have got it. I moved the `igraph_t` part outside the loop as well which I shouldn't have done.